osi_logicallane.proto

```diff
@@ -0,0 +1,502 @@
+    //
+    // Now there are two basic possibilities to define the logical lanes:
+    // - There could be two logical lanes: one encompassing l4, part of l3 and
+    //   l5; another encompassing l2, part of l3 and l1. In this case, one
+    //   logical lane can map to a number of physical lanes, and the mapping
+    //   should include s positions, so it is clear which part of the logical
+    //   lane maps to which physical lane.
+    // - There could be 6 logical lanes: l1, l2, l4, l5 and two lanes for each
+    //   halve of l3. Then each logical lane would map to a single physical
+    //   lane, but logical lanes would have to be split each time a lane
+    //   marking appears or disappears.
+    //
+    optional Identifier physical_lane_id = 4;
+
+    // The reference line for this logical lane
+    //
+    // The reference line is used as a coordinate system on this lane.
+    //
+    // The reference line should roughly have the same shape as the lane, so
+    // that S coordinates continually increase/decrease along the lane. It is
+    // not required that the reference line has the same direction as the
+    // driving direction of the lane.
+    //
+    // Neighbouring lanes are strongly encouraged to use the same
+    // reference_line, so that vehicles that are next to each other on
+    // neighbouring lanes have comparable S positions.
+    //
+    optional Identifier reference_line_id = 5;
+
+    // Start S position of the lane. Must be in range [sStart,sEnd] of the
+    // reference line.
+    //
+    optional double start_s = 6;
+
+    // End S position of the lane. Must be in range [sStart,sEnd] of the
+    // reference line.
+    // end_s > start_s
+    //
+    optional double end_s = 7;
+
+    // Definition of the intended driving direction.
+    //
+    // \c true means driving direction is according to ascending S position.
+    // \c false means driving direction is according to descending S position.
+    //
+    optional bool reference_line_is_driving_direction = 8;
+
+    // Lanes that are directly right of this lane, without gap or overlap.
+    // "Right" is in definition direction (not driving direction), so right lanes
+    // have smaller T coordinates.
+    // Entries must be ordered: first by start_s, then by end_s.
+    //
+    repeated LaneRelation right_adjacent_lane = 9;
+
+    // Lanes that are directly left of this lane, without gap or overlap.
+    // "Left" is in definition direction (not driving direction), so left lanes
+    // have larger T coordinates.
+    // Entries must be ordered: first by start_s, then by end_s.
+    //
+    repeated LaneRelation left_adjacent_lane = 10;
+
+    // Lanes that partially or completely overlap this lane. Only overlaps
+    // laterally larger than 5cm are considered overlaps for the purpose of
+    // this relation.
+    // Entries must be ordered: first by start_s, then by end_s.
+    //
+    repeated LaneRelation overlapping_lane = 11;
```

Comment:
Does this only apply for junctions or, are there other places where this can happen?

Reply:
It depends on the map format. Currently in OpenDRIVE this can only happen in junctions (at least if correctly following the specification). This might change depending on how bike lanes will be defined in OpenDRIVE 1.8 (see https://code.asam.net/simulation/standard/opendrive/-/issues/18).

If using e.g. OpenLaneModel as the underlying map, then restricted lanes (like the above-mentioned bike lines, but also bus lanes) can already overlap other lanes, even outside of junctions.